draft-ietf-httpbis-digest-headers.md

```diff
@@ -358,152 +438,213 @@ Examples:
 ~~~ http-message
 Want-Digest: sha-256
 Want-Digest: sha-512;q=0.3, sha-256;q=1, unixsum;q=0
```

Comment:
As an implementation consideration, both sf-list and sf-dictionary can probably parse want-digest.

```In [2]: d=http_sfv.Dictionary()

In [3]: d.parse(b'sha-512;q=0.3, sha-256;q=1, unixsum;q=0')

In [4]: d.to_json()
Out[4]: 
{'sha-512': [True, {'q': Decimal('0.3')}],
 'sha-256': [True, {'q': 1}],
 'unixsum': [True, {'q': 0}]}
```
or

```
In [5]: d=http_sfv.List()

In [6]: d.parse(b'sha-512;q=0.3, sha-256;q=1, unixsum;q=0')

In [7]: d.to_json()
Out[7]: 
[[{'__type': 'token', 'value': 'sha-512'}, {'q': Decimal('0.3')}],
 [{'__type': 'token', 'value': 'sha-256'}, {'q': 1}],
 [{'__type': 'token', 'value': 'unixsum'}, {'q': 0}]]
```

Reply:
digest-algorithm is a `token`, while list could be a list of `sf-token` and sf-dictionary uses `key`

```
  token          = 1*tchar

     tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
                    / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
                    / DIGIT / ALPHA
                    ; any VCHAR, except delimiters
```
```
sf-token = ( ALPHA / "*" ) *( tchar / ":" / "/" )
```

```
key           = ( lcalpha / "*" )
                   *( lcalpha / DIGIT / "_" / "-" / "." / "*" )
   lcalpha       = %x61-7A ; a-z
```

so there's the possibility that someone has an algorithm identifier that won't actually parse correctly to sf-token or key